vllm/compilation/cuda_graph.py

```diff
@@ -81,8 +79,8 @@ def __init__(self,
         # assert runtime_mode is not NONE(no cudagraph), otherwise, we don't
         # need to initialize a CUDAGraphWrapper.
         assert self.runtime_mode != CUDAGraphMode.NONE
-        if self.graph_pool is None:
-            self.graph_pool = current_platform.get_global_graph_pool()
+        # Always get graph pool from platform
+        self.graph_pool = current_platform.get_global_graph_pool()
```

Comment:
```suggestion
        # TODO: in the future, if we want to use multiple
        # streams, it might not be safe to share a global pool.
        # only investigate this when we use multiple streams
        self.graph_pool = current_platform.get_global_graph_pool()
```